tests/phpunit/tests/option/option.php

```diff
@@ -381,13 +381,17 @@ public function data_option_autoloading() {
 	public function test_update_loosey_options( $old_value, $new_value, $update = false ) {
```

Comment:
Let's take the 28 failures reported by this test.

Most if not all the failing datasets are triggering a query in `update_option()` via `$wpdb->update()`.

Note from the MySQL documentation for UPDATE:
> If you set a column to the value it currently has, MySQL notices this and does not update it.
> [Reference](https://dev.mysql.com/doc/refman/8.0/en/update.html#:~:text=If%20you%20set%20a%20column%20to%20the%20value%20it%20currently%20has%2C%20MySQL%20notices%20this%20and%20does%20not%20update%20it.)

This returns `0` for the number of affected rows and `$wpdb` [increments the value of `wpdb::$num_queries`](https://github.com/wordpress/wordpress-develop/blob/fe71ba1a1b5a833df25323e3a18fbdc4a35c588e/src/wp-includes/class-wpdb.php#L2337).

When `$wpdb->update()` returns `0` for the number of affected rows (via `$wpdb::query()`), `update_option()` [catches this in a falsy check](https://github.com/wordpress/wordpress-develop/blob/fe71ba1a1b5a833df25323e3a18fbdc4a35c588e/src/wp-includes/option.php#L824) and returns `false`.

This reveals the importance of the `get_num_queries()` tests here, as the `assertFalse()` in the tests was providing a false sense of confidence that a database query had not been performed.

Note the return annotation for `update_option()`:
```php
* @return bool True if the value was updated, false otherwise.
```
So when `update_option()` returns `false`, it tells us that the value was not changed. It doesn't tell us whether a query was run.

Reply:
This is a great observation, which wasn't clear by reading the docs for `$wpdb->update()`.